pkg/controller/core.oam.dev/v1alpha2/initializer/initializer_controller.go

```diff
@@ -0,0 +1,195 @@
+/*
+ Copyright 2021. The KubeVela Authors.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+package initializer
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	cpv1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+	"github.com/crossplane/crossplane-runtime/pkg/event"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/klog/v2"
+	"k8s.io/utils/pointer"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/oam-dev/kubevela/apis/core.oam.dev/common"
+	"github.com/oam-dev/kubevela/apis/core.oam.dev/v1beta1"
+	oamctrl "github.com/oam-dev/kubevela/pkg/controller/core.oam.dev"
+	"github.com/oam-dev/kubevela/pkg/oam/discoverymapper"
+	"github.com/oam-dev/kubevela/pkg/oam/util"
+)
+
+// InitializerReconcileWaitTime is the time to wait before reconcile again
+const InitializerReconcileWaitTime = time.Second * 1
+
+// Reconciler reconciles a Initializer object
+type Reconciler struct {
+	client.Client
+	dm                   discoverymapper.DiscoveryMapper
+	Scheme               *runtime.Scheme
+	record               event.Recorder
+	concurrentReconciles int
+}
+
+// Reconcile is the main logic for Initializer controller
+func (r *Reconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
+	klog.InfoS("Reconcile initializer", "initializer", klog.KRef(req.Namespace, req.Name))
+	ctx := context.Background()
+
+	init := new(v1beta1.Initializer)
+	if err := r.Client.Get(ctx, req.NamespacedName, init); err != nil {
+		if apierrors.IsNotFound(err) {
+			err = nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	// TODO(yangsoon) this is a placeholder for finalizer here
+	if init.DeletionTimestamp != nil {
+		return ctrl.Result{}, nil
+	}
+
+	klog.Info("Check the status of the Initializers which you depend on")
+	err := r.checkDependsOn(ctx, req.Namespace, init.Spec.DependsOn)
+	if err != nil {
+		klog.ErrorS(err, "Initializers which you depend on are not ready")
+		r.record.Event(init, event.Warning("Initializers which you depend on are not ready", err))
+		return reconcile.Result{RequeueAfter: InitializerReconcileWaitTime}, util.PatchCondition(ctx, r, init, cpv1alpha1.ReconcileError(err))
```

Comment:
don't patch condtion while requeue in a short time, i will case multiple requeue